base/cvd/cuttlefish/common/libs/utils/archive_test.cpp

```diff
@@ -0,0 +1,90 @@
+//
+// Copyright (C) 2024 The Android Open Source Project
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "common/libs/utils/archive.h"
+
+#include <fstream>
+#include <string>
+#include <vector>
+
+#include "common/libs/utils/files.h"
+#include "common/libs/utils/result.h"
+#include "common/libs/utils/result_matchers.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "android-base/file.h"
+
+namespace cuttlefish {
+
+void CreateTempFileWithText(const std::string& filepath,
+                            const std::string& text) {
+  std::ofstream file(filepath);
+  file << text;
+  file.close();
+}
+
+// Test fixture for common setup and teardown
+class ExtractArchiveContentsTests : public ::testing::Test {
+ protected:
+  void SetUp() override {
+    temp_dir_ = TemporaryDir().path;
+    mkdir(temp_dir_.c_str(), 0755);
+    archive_dir_ = temp_dir_ + "/device-imge-123.zip";
+    mkdir(archive_dir_.c_str(), 0755);
```

Comment:
Use `EnsureDirectoryExists` from `files.h` to create directories and don't ignore the result of calling the function.